service/test/integration2_test.go

```diff
@@ -506,6 +506,80 @@ func TestClientServer_clearBreakpoint(t *testing.T) {
 	})
 }
 
+func TestClientServer_toggleBreakpoint(t *testing.T) {
+	withTestClient2("testtoggle", t, func(c service.Client) {
+
+		toggle := func(bp *api.Breakpoint) {
+			dbp, err := c.ToggleBreakpoint(bp.ID)
+			if err != nil {
+				t.Fatalf("Unexpected error: %v", err)
+			}
+			if dbp.ID != bp.ID {
+				t.Fatalf("The IDs don't match")
+			}
+		}
+
+		// This one is toggled twice
+		bp1, err := c.CreateBreakpoint(&api.Breakpoint{FunctionName: "main.lineOne"})
+		if err != nil {
+			t.Fatalf("Unexpected error: %v\n", err)
+		}
+
+		toggle(bp1)
+		toggle(bp1)
+
+		// This one is toggled once
+		bp2, err := c.CreateBreakpoint(&api.Breakpoint{FunctionName: "main.lineTwo"})
+		if err != nil {
+			t.Fatalf("Unexpected error: %v\n", err)
+		}
+
+		toggle(bp2)
+
+		// This one is never toggled
+		bp3, err := c.CreateBreakpoint(&api.Breakpoint{FunctionName: "main.lineThree"})
+		if err != nil {
+			t.Fatalf("Unexpected error: %v\n", err)
+		}
+
+		if e, a := 3, countBreakpoints(t, c); e != a {
+			t.Fatalf("Expected breakpoint count %d, got %d", e, a)
+		}
+
+		enableCount := 0
+		disabledCount := 0
+
+		contChan := c.Continue()
+		for state := range contChan {
+			if state.CurrentThread != nil && state.CurrentThread.Breakpoint != nil {
+				switch state.CurrentThread.Breakpoint.ID {
+				case bp1.ID, bp3.ID:
+					enableCount++
+				case bp2.ID:
+					disabledCount++
+				}
+
+				t.Logf("%v", state)
+			}
+			if state.Exited {
+				continue
+			}
+			if state.Err != nil {
+				t.Fatalf("Unexpected error during continue: %v\n", state.Err)
+			}
+
+		}
+
+		if enableCount != 2 {
```

Comment:
Currently this test fails here. Although I think there's should be 2 hits it counts only 1. Maybe I don't understand how the test suite performs the execution of the test